mediatek: add support for TP-Link Archer AX80v1(US/RU/CA)

HateTM opened the pull request:
### **This commit adds support for TP-Link Archer AX80v1(US/RU/CA).**

### Device specification

```
SoC Type:   MediaTek MT7986AV, Cortex-A53, 64-bit
RAM:        ESMT M15T4G16256 (512MB)
Flash:      ESMT F50L1G41LB (128 MB)
Ethernet:   MediaTek MT7531AE + 2.5GbE MaxLinear GPY211C0VC (SLNW8)
Ethernet:   1x2.5Gbe (WAN/LAN 2.5Gbps), 4xGbE (WAN/LAN 1Gbps, LAN1, LAN2, LAN3)
WLAN 2g:    MediaTek MT7976GN
WLAN 5g:    MediaTek MT7976AN
LEDs:       1 red,1 green,1 blue status LEDs
Buttons:    4 (Reset,ledswitch,wps,wlan)
USB ports:  1 (USB 3.0)
Power:      12 VDC, 3,3 A
Connector:  Barrel
Bootloader: Main U-Boot - U-Boot 2022.01-rc4. Additionally, both UBI
            slots contain "seconduboot" (also U-Boot 2022.01-rc4)
```
### Installation (UART)

1. Place OpenWrt initramfs image on tftp server with IP 192.168.1.2
2. Attach UART, switch on the router and interrupt the boot process by pressing 'Ctrl-C'
3. Load and run OpenWrt initramfs image:
```
      tftpboot initramfs-kernel.bin
      bootm
```
4. **!!Attention!!** is very important! After entering OpenWrt, please set / update the environment variables:
```
fw_setenv bootargs "ubi.mtd=ubi0 console=ttyS0,115200n1 loglevel=8 earlycon=uart8250,mmio32,0x11002000 init=/etc/preinit" 
fw_setenv mtdids "spi-nand0=spi-nand0"
fw_setenv mtdparts "spi-nand0:2M(boot),1M(u-boot-env),50M(ubi0),50M(ubi1),8M(userconfig),4M(tp_data),8M(mali_data)"
fw_setenv tp_boot_idx 0
```
5. Run 'sysupgrade -n' with the sysupgrade OpenWrt image.

### Recovery

1. Press Reset button and power on the router
2. Navigate to U-Boot recovery web server [http://192.168.1.1](http://192.168.1.1) 
3. Upload the OEM firmware

### Stock layout

```
0x000000000000-0x000000200000 : "boot"
0x000000200000-0x000000300000 : "u-boot-env"
0x000000300000-0x000003500000 : "ubi0"
0x000003500000-0x000006700000 : "ubi1"
0x000006700000-0x000006f00000 : "userconfig"
0x000006f00000-0x000007300000 : "tp_data"
0x000007300000-0x000007B00000 : "mali_data"
```
### Ubi0/Ubi1 format

U-Boot at boot checks that all volumes are in place: 
```
+-------------------------------+
| Volume Name: uboot   Vol ID: 0|
| Volume Name: kernel  Vol ID: 1|
| Volume Name: rootfs  Vol ID: 2|
+-------------------------------+
```
### MAC addresses

```
LAN = label
WAN = label+1
WLAN 2g = label 
WLAN 5g = label-1
```
label MAC address was found in UBI partition "tp_data", file
"default-mac". OEM wireless eeprom is also there (file
"MT7986_EEPROM.bin").

Signed-off-by: HateTM <ichizakurain@gmail.com>

achterin commented:
> 1. Download the transition image - https://github.com/HateTM/download/releases/download/stock/Archer-ax80-oem-sysupgrade.bin

what is needed for a user to build this image themselves?

HateTM commented:
> > 1. Download the transition image - https://github.com/HateTM/download/releases/download/stock/Archer-ax80-oem-sysupgrade.bin
> 
> what is needed for a user to build this image themselves?

In order to build such an image yourself, you will need the TP-Link GPL SDK, which is freely available on the official website.

HateTM commented on target/linux/mediatek/filogic/config-6.6:
```diff
@@ -225,6 +225,8 @@ CONFIG_IRQ_TIME_ACCOUNTING=y
 CONFIG_IRQ_WORK=y
 CONFIG_JBD2=y
 CONFIG_JUMP_LABEL=y
+CONFIG_LEDS_LP55XX_COMMON=y
+CONFIG_LEDS_LP5523=y
```
Need this, because the first filogic device with those LEDs.

hauke replied:
Please move this into a kernel package, see b33fa6ac9065028046016ea631f6be0926d7abd0.
This allows you to add it only to one board and not for all.

HateTM reviewed (commented):
Well, that looks good.

HateTM commented:
Please @hauke re-check.

Demiurge67 reviewed (approved):
It`s very big work.
It`s good work.
I've tested it on my router Archer AX80(RU) and I have to say that everything works perfectly!

yjhn commented:
There are several versions of this router for different regions. Not sure if their internal hardware is identical. Does yours have internal or external antennas?

yjhn commented:
> https://github.com/HateTM/download/releases/download/stock/Archer-ax80-oem-sysupgrade.bin

Gives a 404. Maybe the repo is private?

HateTM commented:
> > https://github.com/HateTM/download/releases/download/stock/Archer-ax80-oem-sysupgrade.bin
> 
> Gives a 404. Maybe the repo is private?

Oh, my mistake! Now it's a public.

Demiurge67 commented:
> ~There are several versions of this router for different regions. Not sure if their internal hardware is identical. Does yours have internal or external antennas?~ Edit: sorry, didn't notice that's specified at the top... US version has external antennas.

US, CA, RU version are identical. EU and JP versions have similar hardware, - 1.6GHz MediaTek MT7986B processor, firmware on the same SDK and more LEDs.

Harwesta reviewed (approved):
Perfect!

hauke commented on target/linux/mediatek/dts/mt7986a-tplink-archer-ax80-v1.dts:
```diff
@@ -0,0 +1,407 @@
+			linux,code = <KEY_RESTART>;
+			gpios = <&pio 7 GPIO_ACTIVE_LOW>;
+			linux,input-type = <1>;
+		};
+		ledswitch {
```
```suggestion
		};

		ledswitch {
```
Please add a empty line after a `};`

hauke commented on target/linux/mediatek/dts/mt7986a-tplink-archer-ax80-v1.dts:
```diff
@@ -0,0 +1,407 @@
+	pinctrl-0 = <&spi_flash_pins>;
+	cs-gpios = <0>, <0>;
+	status = "okay";
+	spi_nand: spi_nand@1 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "spi-nand";
+		reg = <1>;
+		spi-max-frequency = <20000000>;
+		spi-tx-buswidth = <4>;
+		spi-rx-buswidth = <4>;
+		
```
Some Mediatek devices are using `mediatek,nmbm`. Did you check that the original image does not use it? If you are unsure please post a boot log of the vendor firmware.

HateTM replied:
I checked that the original firmware and u-boot bootloader do not use NMBM, just like in Mercusys MR90x.

Rygle commented:
Just curious, is this for the American/Canadian TP-Link Archer AX80, or the European/Asian TP-Link Archer AX80?

See this discussion
[https://community.tp-link.com/en/home/forum/topic/618426](https://community.tp-link.com/en/home/forum/topic/618426)

The American one has external antennas and seems to be the one pictured at the US FCC here (including internal pics);
[https://fccid.io/2AXJ4AX80/Internal-Photos/2206TW0119-Internal-Photograph-6112126](https://fccid.io/2AXJ4AX80/Internal-Photos/2206TW0119-Internal-Photograph-6112126)

The external shots I can find of the European/Asian version with internal antennas seems to indicate that the motherboard is quite different as the power plug and button are in different spots and the spaces between Ethernet ports are also non-existent for the Euro/Asian model.

American/Canadian model
https://m.media-amazon.com/images/I/61NzpxtuqjL.jpg

Euro/Asian Model
https://i.ebayimg.com/images/g/PlEAAOSwW5lk7-7M/s-l1600.jpg

They may well have the same chipsets, but the layout indicates at least physical board differences.

adamscott commented:
@Rygle, it seems it was already answered, or at least, in part.

> > ~There are several versions of this router for different regions. Not sure if their internal hardware is identical. Does yours have internal or external antennas?~ Edit: sorry, didn't notice that's specified at the top... US version has external antennas.
> 
> 
> 
> US, CA, RU version are identical. EU and JP versions have similar hardware, - 1.6GHz MediaTek MT7986B processor, firmware on the same SDK, internal antennas and more LEDs.

@Demiurge67 says that they tested on their RU router. And yes, US, CA, and RU seem identical.

![image](https://github.com/user-attachments/assets/b8078c03-9518-4dad-8b01-6d13795b9548)

adamscott commented:
By the way, I just bought myself a US/CA AX80 (with 8 antennas 🤓) to test and use this PR.

The box is huge. I can't wrap my head around this.

![image](https://github.com/user-attachments/assets/c8d220da-8659-491e-a18b-fe4ef87405d9)

HateTM committed:
kernel: add kmod-leds-lp5523

Add kernel module for LP5523/LP55231 LED driver.
The kmod-leds-lp5523 depends on kmod-leds-lp55xx-common.

Signed-off-by: Sergey Shlukov <ichizakurain@gmail.com>

adamscott commented:
> > > 1. Download the transition image - https://github.com/HateTM/download/releases/download/stock/Archer-ax80-oem-sysupgrade.bin
> > 
> > 
> > what is needed for a user to build this image themselves?
> 
> In order to build such an image yourself, you will need the TP-Link GPL SDK, which is freely available on the official website.

What do you do with the [TP-Link GPL SDK](https://static.tp-link.com/upload/gpl-code/2023/202307/20230727/gpl_ax80.tar.gz) ([available here](https://www.tp-link.com/us/support/download/archer-ax80/v1/#GPL-Code))? It seems to rely on openwrt-21.02. Do I need to use some files from the SDK and compile the PR only?

HateTM commented:
> > > > 1. Download the transition image - https://github.com/HateTM/download/releases/download/stock/Archer-ax80-oem-sysupgrade.bin
> > > 
> > > 
> > > what is needed for a user to build this image themselves?
> > 
> > In order to build such an image yourself, you will need the TP-Link GPL SDK, which is freely available on the official website.
> 
> What do you do with the [TP-Link GPL SDK](https://static.tp-link.com/upload/gpl-code/2023/202307/20230727/gpl_ax80.tar.gz) ([available here](https://www.tp-link.com/us/support/download/archer-ax80/v1/#GPL-Code))? It seems to rely on openwrt-21.02. Do I need to use some files from the SDK and compile the PR only?

You dont need files from SDK, just compile PR with your packages. Upgrade stock firmware to openwrt initramfs from TP-Link  WebUI with transition image then follow instruction.In the end do sysupgrade -n with your sysupgrade image.

adamscott commented:
> You dont need files from SDK, just compile PR with your packages. Upgrade stock firmware to openwrt initramfs from TP-Link WebUI with transition image then follow instruction.In the end do sysupgrade -n with your sysupgrade image.

But how do we build the transition image? Is it just compiling the sdk per itself?

HateTM commented:
> > You dont need files from SDK, just compile PR with your packages. Upgrade stock firmware to openwrt initramfs from TP-Link WebUI with transition image then follow instruction.In the end do sysupgrade -n with your sysupgrade image.
> 
> But how do we build the transition image? Is it just compiling the sdk per itself?

The transition image is built using an app from the GPL SDK. TP-Link did not provide the source code of this application.

Melonpan38 commented:
Hello, this is saving my old router from collecting dust haha. However, have people who have successfully installed openwrt onto the AX80 been able to access LuCI? I seem to only be able to ssh into OpenWRT. I'm using a AX80(CA) V.1 router.

Demiurge67 commented:
> Hello, this is saving my old router from collecting dust haha. However, have people who have successfully installed openwrt onto the AX80 been able to access LuCI? I seem to only be able to ssh into OpenWRT. I'm using a AX80(CA) V.1 router.

Not only Luci is available, but I also made a version for Rooter (for connecting almost any LTE modems). There are no complaints about OpenWRT/Rooter. You can test the firmware by downloading it from here (only Russian translate, but you can to use Gooooooogle): https://4pda.to/forum/index.php?showtopic=1070746&view=findpost&p=135024223

hauke commented on target/linux/mediatek/dts/mt7986a-tplink-archer-ax80-v1.dts:
```diff
@@ -0,0 +1,417 @@
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
```
Is it needed to poll the buttons? It would be better to use interrupt driven buttons if it works:
```
compatible = "gpio-keys";
```

hauke replied:
After doing this change please check that the buttons are still working.

HateTM replied:
The buttons are still working.

hauke commented:
Please order the two commits in the other order. First the one adding the LED kmod and then the commit adding the device.

adamscott commented:
I tested the PR. It works, but needs the supplied transition image. Without instructions to build it, it's kinda weird though and it requires total trust of the uploader.

What will be the end-user experience to install OpenWRT? Will it require that file?

hauke commented:
Why can't OpenWrt build the transition image?

Demiurge67 commented:
> I tested the PR. It works, but needs the supplied transition image. Without instructions to build it, it's kinda weird though and it requires total trust of the uploader.
> 
> What will be the end-user experience to install OpenWRT? Will it require that file?

I don't understand your concern. OpenWRT is an open product and the created intermediate image is completely transparent. As I understand it, it is based on the build guide that is in the SDK. Yes, the SDK uses a fairly old branch of OpenWRT 21.02, but the instructions are perfectly suitable for 23.05.5  - it's enough to adapt them a little. 
For a simple assessment of the transition image, it is enough to look at the manifest, if there is no trust in the manifest - no one prevents you from using more complex tools, for example, binwalk. Do you think that by downloading initramfs-kernel or transition image you can harm your security? How can you do this if you completely replace both the system kernel and the rootfs a few seconds later?
I also tried several times to build transitional images using SDK recipes, but I didn't have enough knowledge to make them bootable from stock firmware. Then I went the other way, using firmware and recipes for Mercusys MR90X. I got a workable hybrid of the Archer AX80 and Mercusys MR90X. And I also built transitional firmware based on the SDK. But I couldn't make a proper image for the AX80(RU/US/CA) that would be perceived by the stock. This was done by comrade HateTM. Many thanks to him for that.
Objectively, you need to study the SDK recipes more deeply and everything will become clear. Now that the HateTM team-mate has done a great job, it's much easier to do this.
Or are you afraid of evil Russian hackers who will steal your big US secret? )))))

Blockr22 commented:
Hi, just wanted to know how long it takes for firmwares to be officially supported by OPENWRT. I can’t find this router on openwrt website. Thnx

HateTM commented:
> Why can't OpenWrt build the transition image?

The tp-link image format is similar to mkimage-2022, it is header + ubi, but it does not fit. I'm currently working on a script that will be able to assemble an image of the desired format.

Rygle commented:
> Hi, just wanted to know how long it takes for firmwares to be officially supported by OPENWRT. I can’t find this router on openwrt website. Thnx

With code in a volunteer project, there is no definite time. It depends on volunteers writing and reviewing code, which means the time is as long as a piece of string (i.e. undetermined until it is finished).

In OpenWrt only developers who have proved themselves over time can commit code, and then it is only when 6 developers have reviewed any code that generally it gets approved and goes in to the snapshot codebase, and then users can load the snapshot builds if desired, but that is very much a beta testing phase. That is likely what will happen with this device. I have run several devices on snapshot builds for months at a time because they did not have stable builds and for the most part that works because OpenWrt does not allow code to get into builds without these reviews to maintain the quality even at the snapshot stage. Other projects can differ and be much less stable because the code is not carefully checked.

Even then, code does not get in to the stable builds unless it proves itself in the snapshot builds, and until a new stable release is made. Sometimes a device may be added to stable in a point revision (24.10.x), or sometimes it will have to wait for a major release (25.x). Security patches and urgent patches may get backported from snapshot versions to stable, but since this can involve a fair amount of code re-factoring (as the snapshot code may have changed a lot from the stable code in many different areas) this can take varying amounts of effort and knowledge. So it depends on urgency, volunteers, interest, knowledge, skill.

If you are skilled and committed you can also fork the whole OpenWrt code base, add in whatever you want to add in (like this device) and make your own builds with whatever changes you want. If you find it works you may even be able to help others get the code into a good state to be included in the snapshots. Or you may be able to find on the forums where someone else has made a snapshot or community build for your device by doing the same, which happens only rarely. Or you may be extremely lucky and find someone (e.g. on a forum) who is interested enough that they are prepared to build it for you, but that is very unlikely. Occasionally people will also go to other non-vanilla OpenWrt branches like ImmortalWrt because only that code base works on the devices that are available in their area (China, etc).

But in many cases you just have to wait... because code is as long as a piece of string.

KenDodman commented:
What is the status on this? 

Is it useable by the average user on a AX80 router? Does it have a webGUI? does someone need to build the image themselves?

HateTM commented:
Now you don't need to download the transition image!

HateTM commented on package/boot/uboot-mediatek/Makefile:
```diff
@@ -874,6 +886,7 @@ UBOOT_TARGETS := \
 	mt7986_jdcloud_re-cp-03 \
 	mt7986_mercusys_mr90x-v1 \
 	mt7986_netcore_n60 \
+  mt7986_tplink_archer-ax80 \
```
done

achterin commented on package/kernel/linux/modules/leds.mk:
```diff
@@ -316,6 +316,23 @@ endef
+endef
+
+define KernelPackage/leds-lp5523/description
+ This option enables support for Texas Instruments LP5523/LP55231
```
wrong indentation

HateTM replied:
correct indentation

achterin replied:
sorry, my bad, you are right. mistook the description define for the package one. sorry to have wasted your time.

Schneider-Azima commented:
Thanks for your work! I borrowed your method of generating seconduboot.

However, I wonder are your firmwares correct after parallel compilation? It seems that there is no clearly defined dependency between IMAGE targets. In the ubi generated by my parallel compilation, the uImage header of seconduboot, which should have content, is actually empty.

I had to give up on making secondubooot an IMAGE target. Instead I created it in a complete function and put it before the `append-ubi` function. This ensured the order of creation. Refer to #18181 for my practice.

HateTM commented:
> Thanks for your work! I borrowed your method of generating seconduboot.
> 
> However, I wonder are your firmwares correct after parallel compilation? It seems that there is no clearly defined dependency between IMAGE targets. In the ubi generated by my parallel compilation, the uImage header of seconduboot, which should have content, is actually empty.
> 
> I had to give up on making secondubooot an IMAGE target. Instead I created it in a complete function and put it before the `append-ubi` function. This ensured the order of creation. Refer to #18181 for my practice.

Thank you for your feedback ! I looked at your PR and decided to optimize mine as well , removing `IMAGES` that are not needed . Now everything looks more compact .

HateTM commented:
I invite @hauke and others to review. Thank you in advance for your feedback.

gnanasreekar commented:
I have a AX80 and i'm completely new to this subject, 

Is the process to install openWrt currently is just download the bin fine and update the firmware with it in the router?

HateTM commented:
> I have a AX80 and i'm completely new to this subject,
> 
> Is the process to install openWrt currently is just download the bin fine and update the firmware with it in the router?

You need to build the webui-factory.bin firmware with the code from this pull request and, if your router's AX80V1 is not the EU or JP versions, you can safely update it via the tp-link web interface.

HateTM commented:
Greetings! What else should I do to add device support?

RobinpCP commented:
> > I have a AX80 and i'm completely new to this subject,
> > Is the process to install openWrt currently is just download the bin fine and update the firmware with it in the router?
> 
> You need to build the webui-factory.bin firmware with the code from this pull request and, if your router's AX80V1 is not the EU or JP versions, you can safely update it via the tp-link web interface.

Could you just build the webui-factory.bin firmware for download?  Thanks a lot.

Gilgamesh-lzq commented:
【自动回复】 您的邮件已收到，谢谢。[Auto-response] Your email have been received, thank you.

HateTM committed:
mediatek: add support for TP-Link Archer AX80v1(US/RU/CA)

This commit adds support for TP-Link Archer AX80v1(US/RU/CA).

Device specification
SoC Type:   MediaTek MT7986AV, Cortex-A53, 64-bit
RAM:        ESMT M15T4G16256 (512MB)
Flash:      ESMT F50L1G41LB (128 MB)
Ethernet:   MediaTek MT7531AE + 2.5GbE MaxLinear GPY211C0VC (SLNW8)
Ethernet:   1x2.5Gbe (WAN/LAN 2.5Gbps), 4xGbE (WAN/LAN 1Gbps, LAN1, LAN2, LAN3)
WLAN 2g:    MediaTek MT7976GN
WLAN 5g:    MediaTek MT7976AN
LEDs:       1 red,1 green,1 blue status LEDs,

Buttons:    4 (Reset,ledswitch,wps,wlan),
USB ports:  1 (USB 3.0)
Power:      12 VDC, 3,3 A
Connector:  Barrel
Bootloader: Main U-Boot - U-Boot 2022.01-rc4. Additionally, both UBI
            slots contain "seconduboot" (also U-Boot 2022.01-rc4)
Serial console (UART)
                            V
+-------+-------+-------+-------+
| +3.3V |  GND  |  TX   |  RX   |
+---+---+-------+-------+-------+
    |
    +--- Don't connect

Installation (UART)
Place OpenWrt initramfs image on tftp server with IP 192.168.1.2

Attach UART, switch on the router and interrupt the boot process by pressing 'Ctrl-C'

Load and run OpenWrt initramfs image:

      tftpboot initramfs-kernel.bin
      bootm

!!Attention!! is very important! After entering OpenWrt, please set / update the environment variables:

fw_setenv bootargs "ubi.mtd=ubi0 console=ttyS0,115200n1 loglevel=8 earlycon=uart8250,mmio32,0x11002000 init=/etc/preinit"
fw_setenv mtdids "spi-nand0=spi-nand0"
fw_setenv mtdparts "spi-nand0:2M(boot),1M(u-boot-env),50M(ubi0),50M(ubi1),8M(userconfig),4M(tp_data),8M(mali_data)"
fw_setenv tp_boot_idx 0

Run 'sysupgrade -n' with the sysupgrade OpenWrt image from console or Luci WebUI.

Recovery

Press Reset button and power on the router
Navigate to U-Boot recovery web server (192.168.1.1) and upload the OEM firmware

Stock layout

0x000000000000-0x000000200000 : "boot"
0x000000200000-0x000000300000 : "u-boot-env"
0x000000300000-0x000003500000 : "ubi0"
0x000003500000-0x000006700000 : "ubi1"
0x000006700000-0x000006f00000 : "userconfig"
0x000006f00000-0x000007300000 : "tp_data"
0x000007300000-0x000007B00000 : "mali_data"

ubi0/ubi1 format
U-Boot at boot checks that all volumes are in place:

+-------------------------------+
| Volume Name: uboot   Vol ID: 0|
| Volume Name: kernel  Vol ID: 1|
| Volume Name: rootfs  Vol ID: 2|
+-------------------------------+
MAC addresses
+---------+-------------------+-----------+
| label   | 00:eb:xx:xx:xx:be | label     |
| LAN     | 00:eb:xx:xx:xx:be | label     |
| WAN     | 00:eb:xx:xx:xx:bf | label+1   |
| WLAN 2g | 00:eb:xx:xx:xx:be | label     |
| WLAN 5g | 00:eb:xx:xx:xx:bd | label-1   |
+---------+-------------------+-----------+
label MAC address was found in UBI partition "tp_data", file "default-mac".
OEM wireless eeprom is also there (file "MT7986_EEPROM.bin").

Signed-off-by: Sergey Shlukov <ichizakurain@gmail.com>

hauke commented on target/linux/mediatek/dts/mt7986a-tplink-archer-ax80-v1.dts:
```diff
@@ -0,0 +1,333 @@
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		poll-interval = <20>;
```
`poll-interval` is not needed for `gpio-keys`, please remove it.

Orion-121 commented:
I'm having some issues getting this working on my (CA) version.
Going from OEM to [Archer-ax80-oem-sysupgrade.bin] works fine, but when I go to the 24.10.2 sysupgrade my network slows way down.
I tried loading through luci and through ssh (following the [fw_setenv] instructions in the op), but haven't been able to get the wifi to pull more than 10mbps down / 1mbps up when I can hit my full 1000/50mbs using OEM firmware.

Orion-121 commented:
> > So, the TP-Link AX80 requires access to the uart interface in order to flash it with the openwrt.
> > I have the AX80V1.6. Is it compatible with the the firmware? I heard that both the 1.6 and 1.0 have the same hardware, but one is made in Vietnam and the other is not. Is that correct?
> > Thanks.
> 
> You are mistaken. Installing OpenWrt does not require access to the UART. Please read the instructions carefully.

As someone who recently spent 4 days on this task, reading the instructions carefully is not the fix here. And I say this as a professional engineer who specializes in process development and deployment. Accurate instructions about loading from the OEM firmware should be added to the OpenWRT site so people stop ending up on this PR after looking for help when they see that the AX80 is "supported".

Following the Forum or Git links from OpenWRT sends you to this PR, and the first few lines of instruction here say:
Installation (UART)
1. Place OpenWrt initramfs image on tftp server with IP 192.168.1.2
2. Attach UART, switch on the router and interrupt the boot process by pressing 'Ctrl-C'
3. Load and run OpenWrt initramfs image

Was I able to load OpenWRT by loading one of the builds linked in this PR from the OEM firmware? Yes. However the PHY/2.5G port issues I mentioned above had me chasing my tail and believing it was because I hadn't followed the instructions correctly, and not due to some esoteric config issue that was ultimately easier to avoid than to correct (in my case).

lazaruswald commented:
> > So, the TP-Link AX80 requires access to the uart interface in order to flash it with the openwrt.
> > I have the AX80V1.6. Is it compatible with the the firmware? I heard that both the 1.6 and 1.0 have the same hardware, but one is made in Vietnam and the other is not. Is that correct?
> > Thanks.
> 
> You are mistaken. Installing OpenWrt does not require access to the UART. Please read the instructions carefully.

Sorry, only recently I found out about the responses.  

Several days ago, I was able to flash the AX80V1.6. Back then, I hooked a  cp21x to  the router’s uart interface in order to  halt the router’s internal boot-up process (needed or not; at the time I assumed it was required). 

In the past, I had routers that did require more convoluted methods in order to be flashed with  openwrt and others that could be flashed with openwrt by simply using the router’s oem web interface. I assumed that the AX80 was  more like the earlier. 

The uart interface did not have a header. Since removing the board in order to add a header for the uart was  time consuming and somewhat risky (antennas were wall soldered to the board), I  simply used 18 gauge wires that did fit more or less snugly into the holes alongside some weight on the wires to increase the pressure+friction. Monitored the boot-up process of the route with minicom, ctr+c to interrupt it, “flashed” the initramfs file and then permanently flashed it with the squashfs.  

I assumed that in order to interrupt the router’s boot-up process (ctr+c) in order to retrieve the temporary openwrt firmware(initramfs) from a tftpd (server)  would require access to the UART interface.
